src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -18,42 +21,66 @@ export interface DialogProps extends ModalProps {
     /** Dialog title */
     title?: string;
     /** Properties of the accept button */
-    acceptButton: ButtonProps;
+    acceptButton?: ButtonProps;
     /** Properties of the cancel button */
     cancelButton?: ButtonProps;
+    /** the context of the buttons in the footer */
+    variant?: Omit<ButtonContext, 'secondary'>;
+    /** closes the dialog */
+    onClose: () => void;
 }
 
 const { block, elem } = bem('Dialog', styles);
 
 export const Dialog: React.FC<DialogProps> = (props) => {
-    const { isOpen, children, title, acceptButton, cancelButton, ...rest } = props;
+    const {
+        isOpen,
+        children,
+        title,
+        acceptButton,
+        cancelButton,
+        variant = 'primary',
+        onClose,
+        ...rest
+    } = props;
 
+    const { ref, height } = useResizeObserver<HTMLDivElement>();
+    const showButtons = !!acceptButton;
     const isConfirm = !!cancelButton;
 
     return (
-        <Modal {...rest} {...block(props)} isOpen={isOpen}>
-            <div {...elem('content', props)} role="alert">
+        <Modal {...rest} {...block('dialog')} isOpen={isOpen}>
```

Comment:
see https://textkernelteam.slack.com/archives/C9W305EH3/p1729606839227849

Reply:
also I think in general you are using block wrongly, we never pass a string to it. The main classname gets refined in the original call to bem (see line 33 here)